tests/api_test.go

```diff
@@ -0,0 +1,218 @@
+		fmt.Println("Issue marshaling")
+	}
+	return string(bodystr)
+}
+
+func load_env() {
+	if err := godotenv.Load("../.env"); err != nil {
+		fmt.Println("Error loading .env file")
+	}
+	fmt.Println(os.Getenv("TEST_UID"))
+	UID = os.Getenv("TEST_UID")
+}
+
+func create_table() bool {
+	query := fmt.Sprintf(`CREATE TEMP TABLE
+		%s (
+			id bigserial NOT NULL,
+			discord_id text NOT NULL,
+			time_created timestamp without time zone NOT NULL DEFAULT now(),
+			last_modified timestamp without time zone NOT NULL DEFAULT now(),
+			content text NOT NULL,
+			status bigint NOT NULL,
+			task_date timestamp without time zone NOT NULL
+		);
+		ALTER TABLE
+			%s
+		ADD
+			CONSTRAINT tasks_pkey PRIMARY KEY (id)
+		`, task_table, task_table,
+	)
+
+	_, err := db.DB.Exec(query)
+	return err == nil
+}
+
+func init_test() {
+	load_env()
+	// Initialize DB
+	db.InitDB()
+	// Create Test Table
+	create_table()
+	// Use the test table
+	db.TaskTable = task_table
+}
+
+func TestPostTask(t *testing.T) {
+	init_test()
+	body := `{"content": "test task", "status": 1, "taskDate": "2023-06-20T03:00:00Z"}`
+	apitest.New().
+		Handler(handler.Router()).
+		Post("/task").
+		Query("uid", UID).
+		JSON(body).
+		Expect(t).
+		Assert(jsonpath.Matches(`$.id`, "1")).
+		Assert(jsonpath.Equal(`$.discordID`, "150708634370703360")).
+		Assert(jsonpath.Matches(`$.status`, "1")).
+		Assert(jsonpath.Equal(`$.content`, "test task")).
+		Assert(jsonpath.Equal(`$.taskDate`, "2023-06-20T03:00:00Z")).
+		Status(http.StatusCreated).
+		End()
+}
+
+func create_task(t *testing.T, content string, status int, task_date string) PostBody {
+	body := PostBody{
+		Content:  content,
+		Status:   status,
+		TaskDate: task_date,
+	}
+
+	apitest.New().
+		Handler(handler.Router()).
+		Post("/task").
+		Query("uid", UID).
+		JSON(body.str()).
+		Expect(t).
+		Status(http.StatusCreated).
+		End()
+	return body
+}
+
+func TestGetTasks(t *testing.T) {
+	init_test()
+	// Post some data
+	var bodies []PostBody
+	bodies = append(bodies, create_task(t, "test task", 1, "2023-06-20T00:00:01Z"))
+	bodies = append(bodies, create_task(t, "random", 0, "2023-03-20T00:00:00Z"))
+	bodies = append(bodies, create_task(t, "aaa", 2, "2025-06-30T01:00:00Z"))
+
+	for i, body := range bodies {
+		apitest.New().
+			Handler(handler.Router()).
+			Get("/tasks").
+			Query("uid", UID).
+			Expect(t).
+			Assert(jsonpath.Matches(fmt.Sprintf(`$[%d].content`, i), body.Content)).
+			Assert(jsonpath.Matches(fmt.Sprintf(`$[%d].status`, i), fmt.Sprint(body.Status))).
+			Assert(jsonpath.Matches(fmt.Sprintf(`$[%d].taskDate`, i), body.TaskDate)).
+			Status(http.StatusOK).
+			End()
+	}
+}
+
+func TestGetTask(t *testing.T) {
+	init_test()
+	// Post some data
+	var bodies []PostBody
+	bodies = append(bodies, create_task(t, "test task", 1, "2023-06-20T00:00:01Z"))
+	bodies = append(bodies, create_task(t, "random", 0, "2023-03-20T00:00:00Z"))
+	bodies = append(bodies, create_task(t, "aaa", 2, "2025-06-30T01:00:00Z"))
```

Comment:
is it possible to pull this code out into a separate function? reused here. in the future if we want more extensive testing it'll be better to define it in one place i think

Reply:
True